hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/state/keyvalue/AccountReadableKVState.java

```diff
@@ -114,19 +119,33 @@ private Account accountFromEntity(Entity entity, final Optional<Long> timestamp)
                 .ethereumNonce(Objects.requireNonNullElse(entity.getEthereumNonce(), 0L))
                 .expirationSecond(TimeUnit.SECONDS.convert(entity.getEffectiveExpiration(), TimeUnit.NANOSECONDS))
                 .expiredAndPendingRemoval(false)
-                .key(parseKey(entity.getKey()))
+                .key(getKey(entity, isSmartContract))
                 .maxAutoAssociations(Objects.requireNonNullElse(entity.getMaxAutomaticTokenAssociations(), 0))
                 .memo(entity.getMemo())
                 .numberAssociations(() -> tokenAccountBalances.get().all())
                 .numberOwnedNfts(getOwnedNfts(entity.getId(), timestamp))
                 .numberPositiveBalances(() -> tokenAccountBalances.get().positive())
                 .receiverSigRequired(entity.getReceiverSigRequired() != null && entity.getReceiverSigRequired())
-                .smartContract(CONTRACT.equals(entity.getType()))
+                .smartContract(isSmartContract)
                 .tinybarBalance(getAccountBalance(entity, timestamp))
                 .tokenAllowances(getFungibleTokenAllowances(entity.getId(), timestamp))
                 .build();
     }
 
+    private Key getKey(final Entity entity, final boolean isSmartContract) {
+        final var key = parseKey(entity.getKey());
+        if (key == null && isSmartContract) {
+            return Key.newBuilder()
+                    .contractID(ContractID.newBuilder()
+                            .shardNum(commonProperties.getShard())
+                            .realmNum(commonProperties.getRealm())
```

Comment:
You can remove CommonProperties

```suggestion
                            .shardNum(entity.getShard())
                            .realmNum(entity.getRealm())
```

Reply:
Removed.